hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/helpers/KeyValueContainerUtil.java

```diff
@@ -367,14 +370,23 @@ private static void populateContainerMetadata(
       kvContainerData.markAsEmpty();
     }
 
+    // Set container data checksum.
+    Long containerDataChecksum = metadataTable.get(
+        kvContainerData.getContainerDataChecksumKey());
+
+    if (containerDataChecksum != null) {
+      kvContainerData.setDataChecksum(containerDataChecksum);
+    } else if (ContainerChecksumTreeManager.hasContainerChecksumFile(kvContainerData)) {
+      populateContainerDataChecksum(kvContainerData, metadataTable);
```

Comment:
`populateContainerDataChecksum` already has a check for the file existing, so I think we can leave `hasContainerChecksumFile` private. Additionally I think `populateContainerMetadata` should be the one writing to the table if the value is present, since it already read from it. This is consistent with how the other counters are updated. `populateContainerDataCheckum` would then just fall back to deserializing the tree if it exists and getting the value from there, and the name can be updated.